GDJS/Runtime/inputmanager.js

```diff
@@ -41,22 +41,52 @@ gdjs.InputManager.MOUSE_RIGHT_BUTTON = 1;
 /** @constant {number} */
 gdjs.InputManager.MOUSE_MIDDLE_BUTTON = 2;
 
+/**
+ * Returns the location-modified keyCode.
+ *
+ * @private
+ */
+gdjs.InputManager.prototype.getModifiedKeyCode = function(keyCode, location) {
+  if (keyCode > 0 && location) {
+    // If it is a numpad number, do not modify it.
+    if (96 <= keyCode && keyCode <= 105) {
+      return keyCode;
+    }
+
+    if (
+      location === KeyboardEvent.DOM_KEY_LOCATION_RIGHT ||
+      location === KeyboardEvent.DOM_KEY_LOCATION_NUMPAD
+    ) {
+      return keyCode + 1000;
+    }
+  }
+
+  // Keys with unspecified locations are treated as standard keys.
+  return keyCode;
+};
+
 /**
  * Should be called whenever a key is pressed
  * @param {number} keyCode The key code associated to the key press.
+ * @param {number} location The location of the event.
```

Comment:
Let's document that 0 = standard key, 1 = left key, 2 = right key, 3 = numpad key.